plugins/quetz_content_trust/quetz_content_trust/api.py

```diff
@@ -29,19 +29,14 @@ def get_db_manager():
         db.close()
 
 
-def post_file(file):
+def post_file(file, channel_name):
     if type(file.file) is SpooledTemporaryFile and not hasattr(file, "seekable"):
         file.file.seekable = file.file._file.seekable
 
     file.file.seek(0, os.SEEK_END)
     file.file.seek(0)
 
-    # channel_name is passed as "" (empty string) since we want to upload the file
-    # in a host-wide manner i.e. independent of individual channels.
-    # this hack only works for LocalStore since Azure and S3 necessarily require
-    # the creation of `containers` and `buckets` (mapped to individual channels)
-    # before we can upload a file there.
-    pkgstore.add_file(file.file.read(), "", file.filename)
+    pkgstore.add_file(file.file.read(), channel_name, file.filename)
 
 
 @router.post("/api/content-trust/private-key")
```

Comment:
I would maybe modify this endpoint to be more explicit about the role targeted, like `/api/content-trust/pkg-mgr/private-key`